src/Notifications/NotificationCenterController.cs

```diff
@@ -78,8 +75,12 @@ internal NotificationCenterController(DynamoView view, DynamoLogger dynLogger)
             notificationUIPopup.webView.EnsureCoreWebView2Async();
             notificationUIPopup.webView.CoreWebView2InitializationCompleted += WebView_CoreWebView2InitializationCompleted;
             logger = dynLogger;
-
-            RequestNotifications();
+            // If user turns on the feature, they will need to restart Dynamo to see the count
+            // This ensures no network traffic when Notification center feature is turned off
+            if (dynamoViewModel.PreferenceSettings.EnableNotificationCenter) 
+            {
+                RequestNotifications();
```

Comment:
As far as I know the RequestNotifications(); only populates the number of notifications in the bell icon but this won't prevent that the webapp consume the REST API (not sure if that is the expected behavior).

Reply:
@RobertGlobant20 This will prevent Dynamo making the REST API calls, do you mean the WebApp will still do it?

Reply:
what about simply not loading the extension?

Reply:
@QilongTang if the objective is to prevent just Dynamo to make REST API calls then is OK but if you want also to prevent those calls in the WebApp then is a different story.

Reply:
@RobertGlobant20 My understanding is that the latest version of Notification Center is using process.env to control the endpoint and REST API calls will be made depending on if the endpoint address exist or not. When consumed by Dynamo, the end point is empty. I believe @filipeotero made such change so when used in Dynamo, the WebApp itself should not pull anything but rely on info passed by Dynamo.

Reply:
@filipeotero Please let me know if my understanding is correct. If not, then happy to follow up

Reply:
@QilongTang You are correct. The webApp itself will not request notifications if `process.env.NOTIFICATION_URL` is not set and it is empty when the app is consumed by dynamo. Dynamo calls the endpoint and passes notifications to `window.setNotifications`.

Reply:
That's correct. In Dynamo, the notification center is waiting to push the notifications. Executing the notification center in development mode will grab the notifications by itself.